src/v/pandaproxy/schema_registry/avro.cc

```diff
@@ -296,7 +302,21 @@ result<void> sanitize(json::Value::Object& o, sanitize_context& ctx) {
             new_namespace = std::move(fullname);
         }
 
-        if (!new_namespace.empty()) {
+        if (!new_namespace.has_value()) {
+            if (auto it = o.FindMember("namespace"); it != o.MemberEnd()) {
+                if (!it->value.IsString()) {
+                    return error_info{
+                      error_code::schema_invalid,
+                      "Invalid JSON Field \"namespace\""};
+                }
+                new_namespace.emplace(
+                  it->value.GetString(), it->value.GetStringLength());
+            }
+        }
+
+        if (new_namespace.has_value() && ctx.ns.top() != new_namespace) {
+            ctx.ns.emplace(*new_namespace);
+            pop_ns.emplace(std::move(pop_ns_impl));
```

Comment:
I do not see `pop_ns` used anywhere else. Since it is an optional, I expected to see something checking the value of `pop_ns`. Is this variable needed? What is its intended use?

Could we simply call stack.pop somewhere towards the end of the method?

Reply:
The condition for push is a little tricky and uses local scoped variables, pop is symmetric; but the scope must exceed this block so that the recursive calls are covered.

I thought an RAII object to control the pop would be simpler than checking every exit. The `deferred_action` has a `release` member which would allow the optional to be removed, but there are a lot of paths that would require that call, somewhat defeating the purpose.